.github/workflows/deploy.yml

```
@@ -51,17 +70,27 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     needs: build
-
-    permissions:
-      pages: write
-      id-token: write
-
+    
     environment:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
+      name: ${{ github.ref == 'refs/heads/main' && 'github-pages' || 'github-pages-preview' }}
+      url: ${{ github.ref == 'refs/heads/main' && steps.deployment.outputs.page_url || format(steps.deployment.outputs.page_url, '/preview', github.event.pull_request.number) }}

```

This should not be needed; the deploy step should output the url it deployed to.

```suggestion
      url: ${{ steps.deployment.outputs.page_url }}
```

If it doesn't output a preview url, that's because it deployed to the main site.